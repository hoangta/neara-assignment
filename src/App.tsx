import { useState, useEffect } from 'react'
import MFAStore from './stores/mfaStore'
import Home from './pages/home'
import Create from './pages/create'
import './App.css'

enum Screen {
  Home,
  Create,
}

const store = new MFAStore()

const App = () => {
  const [screen, setScreen] = useState<Screen>(Screen.Home)

  useEffect(() => {
    store.start()
  }, [])

  const getContent = () => {
    switch (screen) {
      case Screen.Home:
        return <Home store={store} onCreate={() => setScreen(Screen.Create)} />

      case Screen.Create:
        return <Create store={store} onBack={() => setScreen(Screen.Home)} />
    }
  }

  return <div className="app">{getContent()}</div>
}

export default App
