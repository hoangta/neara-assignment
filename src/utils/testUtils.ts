import { MFA } from "../types/mfa";

export const mockStore = () => {
  jest.mock("../stores/mfaStore", () =>
    jest.fn().mockImplementation(() => ({
      mfas: mockMFAs,
      createMFA: jest.fn(),
      switchMFAs: jest.fn(),
    }))
  );
};

export const mockSystemTime = (date: Date) => {
  jest.useFakeTimers("modern");
  jest.setSystemTime(date);
};

export const mockSVGElementAnimate = () => {
  const circleAnimateSpy = jest.fn();
  SVGElement.prototype.animate = circleAnimateSpy;
  return circleAnimateSpy;
};

export const mockMFAs: MFA[] = [
  {
    provider: "Test Provider",
    code: "012345",
    image: "www.testurl.com",
    exp: new Date(2023, 2, 2),
  },
  {
    provider: "Test Provider 2",
    code: "543210",
    image: "www.testurl.com",
    exp: new Date(2023, 2, 3),
  },
];
