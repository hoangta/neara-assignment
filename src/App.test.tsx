import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import { mockStore, mockSVGElementAnimate } from './utils/testUtils'

mockStore();
mockSVGElementAnimate();

describe('App', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<App />)
        expect(asFragment()).toMatchSnapshot()
    })
})