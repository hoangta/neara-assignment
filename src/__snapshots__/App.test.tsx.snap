// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div>
      <div
        class="container"
      >
        <button
          class="button"
          data-testid="edit_button"
        >
          Edit
        </button>
        <div
          class="title"
        >
          Tokens
        </div>
        <button
          class="button"
          data-testid="add_button"
          style="font-size: 25px;"
        >
          ï¼‹
        </button>
      </div>
      <div>
        <div
          class="container"
        >
          <img
            alt="Epic Games"
            class="image"
            src="epicgames.png"
          />
          <div
            class="detail"
          >
            <div
              class="provider"
            >
              Epic Games
            </div>
            <div
              class="code"
            >
              726 759
            </div>
          </div>
          <div
            class="indicatorContainer"
          >
            <svg
              class="indicatorItem"
            >
              <circle
                class="circle"
                cx="20"
                cy="20"
                fill="none"
                r="14"
                stroke="gray"
                stroke-dasharray="87.96459430051421"
                stroke-dashoffset="87.96459430051421"
                stroke-width="2"
              />
            </svg>
            <div
              class="countdownLabel"
            >
              60
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <img
            alt="Google"
            class="image"
            src="google.png"
          />
          <div
            class="detail"
          >
            <div
              class="provider"
            >
              Google
            </div>
            <div
              class="code"
            >
              707 310
            </div>
          </div>
          <div
            class="indicatorContainer"
          >
            <svg
              class="indicatorItem"
            >
              <circle
                class="circle"
                cx="20"
                cy="20"
                fill="none"
                r="14"
                stroke="gray"
                stroke-dasharray="87.96459430051421"
                stroke-dashoffset="87.96459430051421"
                stroke-width="2"
              />
            </svg>
            <div
              class="countdownLabel"
            >
              60
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <img
            alt="Facebook"
            class="image"
            src="facebook.png"
          />
          <div
            class="detail"
          >
            <div
              class="provider"
            >
              Facebook
            </div>
            <div
              class="code"
            >
              603 134
            </div>
          </div>
          <div
            class="indicatorContainer"
          >
            <svg
              class="indicatorItem"
            >
              <circle
                class="circle"
                cx="20"
                cy="20"
                fill="none"
                r="14"
                stroke="gray"
                stroke-dasharray="87.96459430051421"
                stroke-dashoffset="87.96459430051421"
                stroke-width="2"
              />
            </svg>
            <div
              class="countdownLabel"
            >
              60
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <img
            alt="Reddit"
            class="image"
            src="reddit.png"
          />
          <div
            class="detail"
          >
            <div
              class="provider"
            >
              Reddit
            </div>
            <div
              class="code"
            >
              244 445
            </div>
          </div>
          <div
            class="indicatorContainer"
          >
            <svg
              class="indicatorItem"
            >
              <circle
                class="circle"
                cx="20"
                cy="20"
                fill="none"
                r="14"
                stroke="gray"
                stroke-dasharray="87.96459430051421"
                stroke-dashoffset="87.96459430051421"
                stroke-width="2"
              />
            </svg>
            <div
              class="countdownLabel"
            >
              60
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
